virtualizarr/writers/icechunk.py

```diff
@@ -51,7 +59,14 @@ def dataset_to_icechunk(ds: Dataset, store: "IcechunkStore") -> None:
 
     # TODO only supports writing to the root group currently
     # TODO pass zarr_format kwarg?
-    root_group = Group.from_store(store=store)
+    if store.mode.str == "a":
```

Comment:
Just noting that `mode` has disappeared from the latest zarr beta and icechunk main branch.

It is replaced with simply `read_only` as a property on the store. The mode will still exist on the Zarr hierarchy though I believe.

Reply:
🤔 I think we should leave this as-is because it doesn't look like the read_only property is working as I would expect it to in the latest released version of icechunk (`0.1.0a4`):

```python
from icechunk import IcechunkStore, StorageConfig
config = StorageConfig.filesystem("local")

store = IcechunkStore.create(storage=config, read_only=False)
print("Create: " + str(store.mode))
# AccessMode(str='w', readonly=False, overwrite=True, create=True, update=False)

store = IcechunkStore.open_existing(storage=config)
print(store.mode) 
# AccessMode(str='r', readonly=True, overwrite=False, create=False, update=False)

store = IcechunkStore.open_existing(storage=config, read_only=False) 
print(store.mode)
# AccessMode(str='r', readonly=True, overwrite=False, create=False, update=False)

store = IcechunkStore.open_existing(storage=config, mode="a")
print(store.mode)
# AccessMode(str='a', readonly=False, overwrite=False, create=True, update=True)

store = IcechunkStore.open_existing(storage=config, mode="w")
print(store.mode)
# AccessMode(str=w', readonly=False, overwrite=True, create=True, update=False)

store = IcechunkStore.open_existing(storage=config, update=True)
print(store.mode)
# AccessMode(str='r', readonly=True, overwrite=False, create=False, update=False)
```

I note from the above that the only way to open the icechunk store in append mode is still using `mode="a"`

Admittedly, I'm having trouble understanding how all the [AccessMode](https://github.com/earth-mover/icechunk/blob/icechunk-v0.1.0-alpha.4/icechunk/src/zarr.rs#L241-L246) properties are being set in icechunk.

Reply:
@TomNicholas what do you think about pinning the icechunk version and changing the implementation when there is a new release?

Reply:
> it doesn't look like the read_only property is working as I would expect

Even if that's not wrong it's certainly highly counter-intuitive - is it reported upstream?

> pinning the icechunk version and changing the implementation when there is a new release?

That sounds reasonable @abarciauskas-bgse - though hopefully by the time you add documentation to this PR this is fixed upstream 🤞

Reply:
> do you think it is worth reporting

I can't see any issue with the word `read_only` in it, so I just raised it https://github.com/earth-mover/icechunk/issues/404 

In general I feel raising duplicate issues is better than not flagging potentially-undiscovered bugs.

Reply:
I think pinning icechunk is the right thing to do. the `read_only` keyword isnt in the latest released version so its not a concern until then. Sorry for the confusion